pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -995,27 +995,40 @@ def set_continuous_sweep(self):
     # specify the unit, there would be an alternative implementation as a method to allow the user
     # to modify the setting unit without manipulating it via the 'amplitude_unit' property
     display_line = Instrument.control(
-        "DL?", "DL %s",
+        "DL?", "DL %g.11E {amplitude_unit}",
         """
         Control the horizontal display line for use as a visual aid or for
         computational purposes. The default value is 0 dBm.
 
-        Type: :code:`float`, :code:`str`
+        Type: :code:`float`
 
-        Takes a value with the unit of :attr:`amplitude_unit` or 'ON' / 'OFF'
+        Takes a value with the unit of :attr:`amplitude_unit`
 
         .. code-block:: python
 
-            instr.display_line = 'ON'
             instr.display_line = -10
 
-            if instr.detector_mode == 0:
+            if instr.display_line == 0:
                 pass
 
         """,
-        validator=joined_validators(strict_range, strict_discrete_set),
-        values=[[float("-inf"), float("inf")], ["ON", "OFF"]],
-        set_process=lambda v: v if isinstance(v, str) else '%.11E {amplitude_unit}' % v
+        validator=strict_range,
+        values=[float("-inf"), float("inf")]
```

Comment:
Is it useful to _restrict_ the range to -infinity to infinity?
The documentation won`t show this, only the source code.

In this case, I think, the validator and the values could be removed.